File: FlowFactorTodo.API/Entities/TodoTask.cs
using System;

namespace FlowFactorTodo.API.Entities;

public class TodoTask
{
    public int Id { get; set; }
    public required string Title { get; set; }
    public string? Description { get; set; }
    public required string Status { get; set; } = "À faire";
    public int UserId { get; set; }
    public User? User { get; set; }
}

File: flowfactor-ui/src/app/interfaces/todo-task.ts
import { User } from './user';

export interface TodoTask {
  id?: number;
  title: string;
  description?: string;
  status: string;
  user: User;
}
export interface CreateTaskDto {
  title: string;
  description: string;
  status: 'À faire' | 'En cours' | 'Complété';
  user: User;
}

export interface UpdateTaskDto {
  status: 'À faire' | 'En cours' | 'Complété';
}
